queries.py

```diff
@@ -487,37 +475,58 @@ def save_bitcoin_account(bitcoin_account=None):
 
 def is_webhook_valid(url):
     """Tests if a webhook is valid by sending a test payload
-
        Arguments:
-
        url -- Webhook url
     """
     if not url or url == '':
         return False
 
-    payload = {"manage_url": "http://example.com/test/url/for/webhook",
-               "memo": "Congrats! The newly saved webhook works",
-               "additional_data": {
-                   "src_ip": "1.1.1.1",
-                   "useragent": "Mozilla/5.0...",
-                   "referer": "http://example.com/referrer",
-                   "location": "http://example.com/location"
-               },
-               "channel": "HTTP",
-               "time": datetime.datetime.now().strftime('%Y-%m-%d %T') }
-    try:
-        response = requests.post(url,
-                                 simplejson.dumps(payload),
-                                 headers={'content-type': 'application/json'},
-                                 timeout=10)
-        response.raise_for_status()
-        return True
-    except requests.exceptions.Timeout as e:
-        log.err('Timed out sending test payload to webhook: {url}'.format(url=url))
-        return False
-    except requests.exceptions.RequestException as e:
-        log.err('Failed sending test payload to webhook: {url} with error {error}'.format(url=url,error=e))
-        return False
+    slack = "https://hooks.slack.com"
+
```

Comment:
There's a last cleanup of this chunk which shrinks the code size a little. Both POSTs are identical except for their payload, so we can combine them. Something like this will factor out the common network code:
```
if (slack in url):
    payload = {'text': 'Validating new canarytokens webhook'}
else:
    payload = {"manage_url": "http://example.com/test/url/for/webhook",
                "memo": "Congrats! The newly saved webhook works",
                "additional_data": {
                    "src_ip": "1.1.1.1",
                    "useragent": "Mozilla/5.0...",
                    "referer": "http://example.com/referrer",
                    "location": "http://example.com/location"
                },
            "channel": "HTTP",
            "time": datetime.datetime.now().strftime('%Y-%m-%d %T') }
try:
    response = requests.post(url,
                                simplejson.dumps(payload),
                                headers={'content-type': 'application/json'},
                                timeout=10)
    response.raise_for_status()
    return True
except requests.exceptions.Timeout as e:
    log.err('Timed out sending test payload to webhook: {url}'.format(url=url))
    return False
except requests.exceptions.RequestException as e:
    log.err('Failed sending test payload to webhook: {url} with error {error}'.format(url=url,error=e))
    return False
```

Reply:
fixed